app/models/order_detail.rb

```diff
@@ -126,6 +126,8 @@ def journal_or_statement_date
       .where("products.type" => product_type)
   }
 
+  scope :non_cancelled, -> { joins(:order_status).where.not(order_statuses: { name: "Canceled" }) }
```

Comment:
I'm actually surprised we don't have a scope for this already. We should probably be using the `state` column for this rather than doing a join to `OrderStatus`. 

When we do need to compare to an order status, we usually use something like `OrderStatus.canceled_status` (you might also see `OrderStatus.complete.first` scattered throughout the code, which is the older way), so this would probably be more like`where.not(order_status: OrderStatus.canceled_status)`.

Also, I know you hate it, but we settled on `canceled`. There used to be both throughout the code, but a couple years ago we decided we wanted consistency (#173).